R/pkg/R/SQLContext.R

```diff
@@ -197,17 +197,40 @@ writeToFileInArrow <- function(fileName, rdf, numPartitions) {
   }
 }
 
-checkTypeRequirementForArrow <- function(dataHead, schema) {
-  # Currenty Arrow optimization does not support raw for now.
-  # Also, it does not support explicit float type set by users. It leads to
-  # incorrect conversion. We will fall back to the path without Arrow optimization.
-  if (any(sapply(dataHead, is.raw))) {
-    stop("Arrow optimization with R DataFrame does not support raw type yet.")
```

Comment:
what happened to the raw check - not sure I see the replacement

Reply:
I tried to merge that code path by checking `BinaryType`. Previously here I had to check if the first line of R data frame has `raw` or not because I could not have the schema here.

After I tweaked here to have the schema first here in Arrow path, I can now check it by `structType`, not the first line of R data frame.

Reply:
ok got it